assets/tests/components/map.test.tsx

```diff
@@ -293,6 +298,90 @@ describe("<MapFollowingPrimaryVehicles />", () => {
     expect(onClick).not.toHaveBeenCalled()
   })
 
+  test("starts out with no dropdown menu", async () => {
+    jest.mocked(getTestGroups).mockReturnValue([TestGroups.DetoursPilot])
+
+    const vehicle = vehicleFactory.build({})
+    const onClick = jest.fn()
+    const { container } = render(
+      <MapFollowingPrimaryVehicles
+        vehicles={[vehicle]}
+        onPrimaryVehicleSelect={onClick}
+      />
+    )
+
+    expect(
+      container.querySelector(".c-dropdown-popup-wrapper")
+    ).not.toBeInTheDocument()
+  })
+
+  test("brings up a dropdown menu on right-click", async () => {
+    jest.mocked(getTestGroups).mockReturnValue([TestGroups.DetoursPilot])
+
+    const vehicle = vehicleFactory.build({})
+    const onClick = jest.fn()
+    const { container } = render(
+      <MapFollowingPrimaryVehicles
+        vehicles={[vehicle]}
+        onPrimaryVehicleSelect={onClick}
+      />
+    )
+
+    await userEvent.pointer({
+      keys: "[MouseRight>]",
+      target: screen.getByText(runIdToLabel(vehicle.runId!)),
+    })
+
+    expect(
+      container.querySelector(".c-dropdown-popup-wrapper")
+    ).toBeInTheDocument()
+  })
+
+  test("does not bring up a dropdown menu on right-click if the user is not in the right test group", async () => {
+    jest.mocked(getTestGroups).mockReturnValue([])
+
+    const vehicle = vehicleFactory.build({})
+    const onClick = jest.fn()
+    const { container } = render(
+      <MapFollowingPrimaryVehicles
+        vehicles={[vehicle]}
+        onPrimaryVehicleSelect={onClick}
+      />
+    )
+
+    await userEvent.pointer({
+      keys: "[MouseRight>]",
+      target: screen.getByText(runIdToLabel(vehicle.runId!)),
+    })
+
+    expect(
+      container.querySelector(".c-dropdown-popup-wrapper")
+    ).not.toBeInTheDocument()
```

Comment:
suggestion/nit(non-blocking): would using `getByRole` and `queryByRole` instead of `container.querySelector` work for these tests?

When I was looking at the accessibility tree for this, the dropdown shows up as a button, so maybe this would work?
```
expect(screen.getByRole("button", { name: "Start a detour on route"})).toBeVisible()
```
and
```
expect(screen.queryByRole("button", { name: "Start a detour on route"})).not.toBeInTheDocument()
```

Reply:
Yeah good call - I like that a lot better!

https://github.com/mbta/skate/pull/2373/commits/f64e7ebda4a250a47913dcb9f9fca76e82f13450